siliconcompiler/tools/yosys/yosys.py

```diff
@@ -58,124 +62,113 @@ def setup(chip):
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'map_adders', 'False/path to map_adders, techmap adders in Yosys', field='help')
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'synthesis_corner', 'Timing corner to use for synthesis', field='help')
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'dff_liberty', 'Liberty file to use for flip-flop mapping, if not specified the first in the logiclib is used', field='help')
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'abc_constraint_driver', 'Buffer that drives the abc techmapping, defaults to first buffer specified', field='help')
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'abc_constraint_load', 'Capacitive load for the abc techmapping in fF, if not specified it will not be used', field='help')
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'abc_clock_period', 'Clock period to use for synthesis in ps, if more than one clock is specified, the smallest period is used.', field='help')
-            chip.set('tool', tool, 'task', task, 'var', step, index, 'abc_clock_derating', 'Used to derate the clock period to further constrain the clock, values between 0 and 1', field='help')
-
-        elif task == 'fpga':
-            chip.set('tool', tool, 'task', task, 'script', step, index, 'sc_syn.tcl', clobber=False)
-            chip.set('tool', tool, 'task', task, 'input', step, index, design + '.v')
-            chip.set('tool', tool, 'task', task, 'output', step, index, design + '.vg')
-            chip.add('tool', tool, 'task', task, 'output', step, index, design + '_netlist.json')
-            chip.add('tool', tool, 'task', task, 'output', step, index, design + '.blif')
-
-            chip.add('tool', tool, 'task', task, 'require', step, index, ",".join(['fpga', 'partname']))
+    # common to all
+    chip.set('tool', tool, 'task', task, 'option', step, index, '-c', clobber=False)
+    chip.set('tool', tool, 'task', task, 'refdir', step, index, refdir, clobber=False)
+    chip.set('tool', tool, 'task', task, 'regex', step, index, 'warnings', "Warning:", clobber=False)
+    chip.set('tool', tool, 'task', task, 'regex', step, index, 'errors', "^ERROR", clobber=False)
+    for metric in ('cells', 'nets', 'pins'):
+        chip.set('tool', tool, 'task', task, 'report', step, index, metric, "reports/stat.json")
+    for metric in ('cellarea', 'errors', 'warnings', 'cellarea', 'drvs', 'coverage', 'security',
+                   'luts', 'dsps', 'brams', 'registers', 'buffers'):
+        chip.set('tool', tool, 'task', task, 'report', step, index, metric, f"{step}.log")
+
+    # Generic ASIC / FPGA mode setup.
+    mode = chip.get('option', 'mode')
+    if mode == 'asic':
+        setup_asic(chip, task)
+    elif mode == 'fpga':
+        setup_fpga(chip, task)
```

Comment:
probably will want the flow to specify `syn_fpna` and `syn_asic` instead of this switch.

Reply:
The ASIC/FPGA setup was independent of which task was being run, though; they also apply to `lec`.